parser/charset/encoding.go

```diff
@@ -150,6 +183,9 @@ func (e *Encoding) Decode(dest, src []byte) ([]byte, error) {
 
 // DecodeString convert a string from a specific charset to utf-8 charset.
 func (e *Encoding) DecodeString(src string) (string, error) {
+	if e.name == encodings[CharsetUTF8].name {
```

Comment:
WIll an extra check cause a performance regression?

Reply:
DecodeString will called by the lexer.

Reply:
I think it will. Maybe #30288 is a better way to do this. I will disscuss with @tangenta later.

Reply:
@tangenta @Defined2014 Do you still want to go on with this issue, or leave it with #30288?

Reply:
Sorry, I will close it.